packages/virtual-core/src/index.ts

```diff
@@ -76,7 +76,7 @@ export const observeElementRect = <T extends Element>(
 
   const observer = new ResizeObserver((entries) => {
     const entry = entries[0]
-    if (entry) {
+    if (entry?.borderBoxSize?.[0]) {
```

Comment:
Maybe checking if `borderBoxSize` exist in entry would be enough?
```suggestion
    if (entry?.borderBoxSize) {
```

Reply:
@zi-gae can you update this, checking if it exist will be way faster.

Reply:
thank you for your suggestion. https://github.com/TanStack/virtual/pull/518/commits/242ca34fdff518b7bb553620286548d2571e93a7